### PR TITLE
Update deps

### PR DESCRIPTION
**Pull request by sharksforarms**
Update all dependencies

### CODE REVIEW

**Comment by wcampbell0x2a** [comment]
@sharksforarms looks like some failure.

Also, I think you need to add a codecov ci token to fix the coverage step.

```
error: failed to select a version for `env_logger`.
    ... required by package `deku v0.16.0 (/home/runner/work/deku/deku)`
versions that meet the requirements `^0.11.3` are: 0.11.3

the package `deku` depends on `env_logger`, with features: `anstream` but `env_logger` does not have these features.
 It has an optional dependency with that name, but that dependency uses the "dep:" syntax in the features table, so it does not have an implicit feature with that name.
```

**Comment by sharksforarms** [comment]
@wcampbell0x2a updated the msrv to 1.71 (same as env_logger), do you see this being an issue?

I see `1.75.0` in my apt repos, I suspect 1.71 is ok.

File: Cargo.toml [hunk 1]
```diff
@@ -9,7 +9,7 @@ keywords = ["deku", "bits", "serialization", "deserialization", "struct"]
 categories = ["encoding", "parsing", "no-std"]
 description = "bit level serialization/deserialization proc-macro for structs"
 readme = "README.md"
-rust-version = "1.67.1"
+rust-version = "1.71"
```
Comment (wcampbell0x2a) [inline Cargo.toml:12]: This is also noted in the README. And I would expect a note in the CHANGELOG.

**Review by wcampbell0x2a — CHANGES_REQUESTED** [review]

File: Cargo.toml (hunk 1 above)
Comment (sharksforarms) [reply Cargo.toml:12]: done in https://github.com/sharksforarms/deku/pull/438/commits/c2e2de945949bd54d6d9e06f62984195de8df162

**Review by wcampbell0x2a — APPROVED** [review]

**Pull request merged**